server/src/server/session/TransactionOLTP.java

```diff
@@ -982,17 +955,17 @@ private void validateGraph() throws InvalidKBException {
     }
 
     /**
-     * Creates a new shard for the concept
+     * Creates a new shard for the concept - only used in tests
      *
      * @param conceptId the id of the concept to shard
      */
+    @VisibleForTesting
     public void shard(ConceptId conceptId) {
         ConceptImpl type = getConcept(conceptId);
         if (type == null) {
-            LOG.warn("Cannot shard concept [" + conceptId + "] due to it not existing in the graph");
-        } else {
-            type.createShard();
+            throw new RuntimeException("Cannot shard concept [" + conceptId + "] due to it not existing in the graph");
```

Comment:
given that this method is only used in tests, it makes more sense to throw an exception if something is wrong rather than `LOG.warn`